### DISCUSSION TITLE
Nest-access-control and us

### DISCUSSION THREAD

**Discussion by tupe12334** [amplication/amplication#5398] [Polls]
## current

As of today, we are using a package called [`nest-access-control`](https://github.com/nestjsx/nest-access-control).
This package has a huge technical debt and is poorly maintained.
The problem with this is that we are using this package in the code we generate for our users.

I would love to hear what you think we need to do in the **Long** term.

## options

1. Fork it to our package and be responsible for it
2. Maintain this package for them and be dependent on them
3. Remove the package and implement the code in the code that we generate

**Comment by yuval-hazaz** [comment]
I would say that optimally the entire access control module should be built with a plugin so it can be interchanged easily, and the community will be able to create different plugins with other technologies or services. 
For example, I know that @MichaelSolati is working on a demo project that uses permit.io for access control

**Comment by abrl91** [comment]
What is the effort for option 3?

**Reply by tupe12334** [reply]
Not a lot, after all, its an open source package
